addons/account_peppol/models/account_edi_proxy_user.py

```diff
@@ -88,6 +115,59 @@ def _call_peppol_proxy(self, endpoint, params=None):
 
         return response
 
+    def _mark_connection_out_of_sync(self):
+        self.ensure_one()
+        if self.peppol_is_token_out_of_sync:
+            return
+        self.write({
+            'peppol_is_token_out_of_sync': True,
+            'refresh_token': None,
+        })
+        response = self._make_request__(
+            f'{self._get_server_url()}/api/peppol/1/mark_connection_out_of_sync',
+            params={'token_desync_counter': self.peppol_token_sync_version},
+            auth_type='asymmetric'
+        )
+        if response.get('error'):
+            raise AccountEdiProxyError(
+                response['error'].get('code', 'unknown_error'),
+                response['error'].get('message', "An unknown error occurred while authenticating with IAP server.")
+            )
+
+    def _peppol_out_of_sync_reconnect_this_database(self):
+        self.ensure_one()
+        assert self.peppol_is_token_out_of_sync
+        self.peppol_token_sync_version += 1
+        response = self._make_request__(
+            f'{self._get_server_url()}/api/peppol/1/resync_connection',
+            params={'token_desync_counter': self.peppol_token_sync_version},
+            auth_type='asymmetric'
+        )
+        if response.get('error'):
+            raise AccountEdiProxyError(
+                response['error'].get('code', 'unknown_error'),
+                response['error'].get('message', "An unknown error occurred while authenticating with IAP server.")
+            )
+        self.write({
+            'refresh_token': response['refresh_token'],
+            'peppol_is_token_out_of_sync': False,
+        })
+
+        # trigger participant status update after resync to confirm token & keep state in sync
+        # but run async, since sync may confirm token server-side (thus increment peppol_token_sync_version)
+        # yet fail before commit, leaving unrecoverable state
+        self.env.ref('account_peppol.ir_cron_peppol_get_participant_status')
+
+    def _peppol_out_of_sync_disconnect_this_database(self):
+        self.ensure_one()
+        assert self.peppol_is_token_out_of_sync
+        # delete this record and company's proxy state
+        self.company_id.write({
+            'account_peppol_proxy_state': 'not_registered',
+            'account_peppol_migration_key': False,
```

Comment:
We added a small helper with Yosua (will be merged soon) that could be relevant here.
https://github.com/odoo/odoo/pull/216864/files#diff-497ad3e52d7f80736673475e27c81d956ef051decee41346cec01a100ff08afcR413